design-toolkit/components/src/components/options/index.tsx

```diff
@@ -12,86 +12,185 @@
 
 'use client';
 import 'client-only';
+import { mergeProps } from '@react-aria/utils';
 import { createContext } from 'react';
 import {
   Header as AriaHeader,
   ListBox as AriaListBox,
   Collection as AriaListBoxCollection,
-  type ListBoxProps as AriaListBoxProps,
   ListBoxSection as AriaListBoxSection,
-  type ListBoxSectionProps as AriaListBoxSectionProps,
+  type ContextValue,
+  composeRenderProps,
+  useContextProps,
 } from 'react-aria-components';
+import 'client-only';
+import { useContext, useMemo } from 'react';
 import {
-  type IOptionsItem,
-  OptionsItem,
-  type OptionsItemProps,
-} from '../options-item';
+  ListBoxItem as AriaListBoxItem,
+  Text as AriaText,
+} from 'react-aria-components';
+import { Icon } from '../icon';
 
-export interface OptionsProps<T extends IOptionsItem>
-  extends AriaListBoxProps<T> {
-  className?: string;
-  description?: string;
-  errorMessage?: string;
-  label?: string;
-  placeholder?: string;
-  size?: OptionsItemProps<T>['size'];
-  type?: OptionsItemProps<T>['type'];
-}
+import type {
+  IOptionsItem,
+  OptionsItemIconProps,
+  OptionsItemProps,
+  OptionsItemTextProps,
+  OptionsProps,
+  OptionsSectionProps,
+} from './types';
+
+import { OptionsStyles } from './styles';
+
+const { list, section, sectionHeader, item, itemIcon, itemContent } =
+  OptionsStyles();
 
 export const OptionsContext = createContext<
-  Pick<OptionsProps<IOptionsItem>, 'size' | 'type'>
+  ContextValue<OptionsProps<IOptionsItem>, HTMLDivElement>
 >({
   size: 'large',
-  type: 'default',
+  color: 'default',
 });
 
 export function Options<T extends IOptionsItem>({
-  children,
-  className,
-  description,
-  errorMessage,
-  label,
-  placeholder,
-  size,
-  type,
+  ref,
   ...props
 }: OptionsProps<T>) {
+  [props, ref] = useContextProps(props, ref ?? null, OptionsContext);
+  const {
+    children,
+    className,
+    description,
+    errorMessage,
+    label,
+    placeholder,
+    size,
+    color,
+    ...rest
+  } = props;
   return (
-    <OptionsContext.Provider value={{ size, type }}>
-      <AriaListBox<T>
-        className='max-h-[200px] overflow-y-auto overflow-x-clip rounded-medium bg-surface-overlay shadow-elevation-overlay outline outline-static-light'
-        {...props}
-      >
+    <OptionsContext.Provider value={{ size, color }}>
```

Comment:
If you implement `size` as a data attribute https://github.com/gohypergiant/standard-toolkit/blob/c16b4797d7ca4d3e081d72ae03a9fec2e580659c/design-toolkit/components/src/components/button/index.tsx#L69 you can use it as a variant selector https://github.com/gohypergiant/standard-toolkit/blob/c16b4797d7ca4d3e081d72ae03a9fec2e580659c/design-toolkit/components/src/components/button/styles.ts#L24